src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp

```diff
@@ -0,0 +1,541 @@
+    gr::transverse_projection_operator(make_not_null(&spatial_projection_Ij),
+                                       unit_interface_normal_vector,
+                                       unit_interface_normal_one_form);
+
+    // Weyl propagating mode
+    gr::weyl_propagating(make_not_null(&weyl_prop_minus), ricci_3,
+                         extrinsic_curvature, inverse_spatial_metric,
+                         cov_deriv_ex_curv, unit_interface_normal_vector,
+                         spatial_projection_IJ, spatial_projection_ij,
+                         spatial_projection_Ij, -1);
+
+    if constexpr (mu_phys != 0.) {
+      auto& weyl_prop_plus =
+          get<::Tags::Tempii<3, VolumeDim, Frame::Inertial, DataType>>(
+              local_buffer);
+      gr::weyl_propagating(make_not_null(&weyl_prop_plus), ricci_3,
+                           extrinsic_curvature, inverse_spatial_metric,
+                           cov_deriv_ex_curv, unit_interface_normal_vector,
+                           spatial_projection_IJ, spatial_projection_ij,
+                           spatial_projection_Ij, 1);
+
+      for (size_t a = 0; a <= VolumeDim; ++a) {
+        for (size_t b = a; b <= VolumeDim; ++b) {
+          for (size_t i = 0; i < VolumeDim; ++i) {
+            for (size_t j = 0; j < VolumeDim; ++j) {
+              U3p.get(a, b) += 2. * projection_Ab.get(i + 1, a) *
+                               projection_Ab.get(j + 1, b) *
+                               weyl_prop_plus.get(i, j);
+              U3m.get(a, b) += 2. * projection_Ab.get(i + 1, a) *
+                               projection_Ab.get(j + 1, b) *
+                               weyl_prop_minus.get(i, j);
+            }
+          }
+        }
+      }
+    } else {
+      // No need to compute U3p or weyl_prop_plus in this case
+      for (size_t a = 0; a <= VolumeDim; ++a) {
+        for (size_t b = a; b <= VolumeDim; ++b) {
+          for (size_t i = 0; i < VolumeDim; ++i) {
+            for (size_t j = 0; j < VolumeDim; ++j) {
+              U3m.get(a, b) += 2. * projection_Ab.get(i + 1, a) *
+                               projection_Ab.get(j + 1, b) *
+                               weyl_prop_minus.get(i, j);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  // Impose physical boundary condition
+  if constexpr (gamma2_in_phys) {
+    DataType tmp(get_size(get<0>(unit_interface_normal_vector)));
+
+    for (size_t c = 0; c <= VolumeDim; ++c) {
+      for (size_t d = c; d <= VolumeDim; ++d) {
+        for (size_t a = 0; a <= VolumeDim; ++a) {
+          for (size_t b = 0; b <= VolumeDim; ++b) {
+            tmp = get<0>(unit_interface_normal_vector) *
+                  three_index_constraint.get(0, a, b);
+            for (size_t i = 1; i < VolumeDim; ++i) {
+              tmp += unit_interface_normal_vector.get(i) *
+                     three_index_constraint.get(i, a, b);
+            }
+            tmp *= get(gamma2);
+
+            bc_dt_v_minus->get(c, d) +=
```

Comment:
[nitpick] it would be nicer to label the iteration indices here to match the documentation (so exchange ab <-> cd) (sane in the `else` loop)

Reply:
fixed